tests/chainer_tests/functions_tests/loss_tests/test_triplet.py

```diff
@@ -89,6 +95,30 @@ def test_backward_gpu_no_cudnn(self):
         self.check_backward(cuda.to_gpu(self.a), cuda.to_gpu(self.p),
                             cuda.to_gpu(self.n), cuda.to_gpu(self.gy))
 
+    def check_double_backward(self, a_data, p_data, n_data, gy_data, gga_data,
+                              ggp_data, ggn_data):
+        def f(a, p, n):
+            y = functions.triplet(
+                a, p, n, margin=self.margin, reduce=self.reduce)
+            return y * y
```

Comment:
Is triplet loss linear? I suppose that this squaring is not needed.

Reply:
Thanks, fixed!